### DISCUSSION TITLE
How to force the usage of 'mem_io_manager' in a @graph_asset

### DISCUSSION THREAD

**Discussion by chrishiste** [dagster-io/dagster#19050] [Q&A]
Hey there,

I'm dealing with a big, multi-partitioned asset that's using the multiprocessor executor and `gcs_pickle_io_manager` (running in k8s), and it's super slow and pretty much stuck because I have few thousand steps with `DynamicOutput`s. When I run it locally with an `in_process` executor, it's quick.

I'm struggling to find a way to set up the executor and IO Manager for a `@graph_asset`. I need to use it since I'm relying on auto materialization.

Also, it seems I can only use one `Definition`, so I can't tweak the executor settings there. I need the multiprocessor for everything else most of the time.

Any idea how can I force the use of `in_process` and `mem_io_manager`?

Using the following doesn't force using the `mem_io_manager`.
```
execution:
  config:
    in_process:
```

Related thread that would potentially fix my issue: 
https://github.com/dagster-io/dagster/issues/4041
https://github.com/dagster-io/dagster/issues/6834

**Comment by chrishiste** [comment]
@alangenfeld Sorry to tag you on this but do you know by any chance if there is way?

**Reply by alangenfeld** [reply]
I believe you can solve this by setting `io_manager_key` to something else and then providing the mem io manager as a resource on that key

https://docs.dagster.io/concepts/io-management/io-managers#per-asset-io-manager

**Reply by chrishiste** [reply]
@alangenfeld Correct me if I'm wrong but `io_manager_key` doesn't look implement on @graph_asset. And using a @graph would require `define_asset_job()` which itself is not compatible with auto-materialization. Is that expected?

I'm on 1.5.13.

**Reply by alangenfeld** [reply]
you would need to set this on the `@op`s that are used within the `@graph_asset`

**Reply by chrishiste** [reply]
Thanks for the answer.

I spent a while trying to figure out what was wrong even with another IO Manager.

I was using the [Dagster concurrency tag](https://docs.dagster.io/guides/limiting-concurrency-in-data-pipelines#limiting-opasset-concurrency-across-runs), it appears that it adds a enormous amount of preparation time for steps. I assume it's due to postgres. A bit unfortunate but I won't be able to use that until performance are acceptable.

I also moved to use the filesystem IO manager because reading and writing to buckets was also adding way more delay than acceptable (this one is outside Dagster's control but I would love a shared memory IO manager). Overall it's still way longer than in process but that's fine for my use case.
